src/main/java/com/iexec/worker/docker/ComputationService.java

```diff
@@ -113,170 +159,217 @@ public boolean runNonTeeComputation(TaskDescription taskDescription,
-        System.out.println(appExecutionResult.getStdout());
-
+    private DockerExecutionResult runTeePostCompute(String secureSessionId, TaskDescription taskDescription) {
+        String chainTaskId = taskDescription.getChainTaskId();
+        List<String> sconeUploaderEnv = sconeTeeService.buildSconeDockerEnv(secureSessionId + "/post-compute",
+                publicConfigService.getSconeCasURL(), "3G");
+        Map<String, String> bindPaths = new HashMap<>();
+        bindPaths.put(workerConfigService.getTaskIexecOutDir(chainTaskId), FileHelper.SLASH_IEXEC_OUT);
+        bindPaths.put(workerConfigService.getTaskOutputDir(chainTaskId), FileHelper.SLASH_OUTPUT);
         DockerExecutionConfig teePostComputeExecutionConfig = DockerExecutionConfig.builder()
                 .chainTaskId(chainTaskId)
                 .containerName(getTaskTeePostComputeContainerName(chainTaskId))
-                .imageUri(teePostComputeImage)
-                .maxExecutionTime(maxExecutionTime)
+                .imageUri(taskDescription.getTeePostComputeImage())
+                .maxExecutionTime(taskDescription.getMaxExecutionTime())
                 .env(sconeUploaderEnv)
-                .bindPaths(getSconeBindPaths(chainTaskId))
+                .bindPaths(bindPaths)
                 .isSgx(true)
                 .build();
         DockerExecutionResult teePostComputeExecutionResult = customDockerClient.execute(teePostComputeExecutionConfig);
-        if (!teePostComputeExecutionResult.isSuccess()) {
-            log.error("Failed to process post-compute on result [chainTaskId:{}]", chainTaskId);
-            return false;
-        }
+        // TODO: remove
         System.out.println("****** Tee post-compute");
         System.out.println(teePostComputeExecutionResult.getStdout());
+        return teePostComputeExecutionResult;
+    }
 
+    private boolean runNonTeePostCompute(TaskDescription taskDescription) {
+        // create /output/iexec_out.zip
+        String chainTaskId = taskDescription.getChainTaskId();
+        String iexecOutPath = workerConfigService.getTaskIexecOutDir(chainTaskId);
+        String saveIn = workerConfigService.getTaskOutputDir(chainTaskId);
+        ResultUtils.zipIexecOut(iexecOutPath, saveIn);
+        // copy /output/iexec_out/computed.json to /output/computed.json
+        // to have the same workflow as TEE.
+        String source = workerConfigService.getTaskIexecOutDir(chainTaskId) + IexecFileHelper.SLASH_COMPUTED_JSON;
+        String target = workerConfigService.getTaskOutputDir(chainTaskId) + IexecFileHelper.SLASH_COMPUTED_JSON;
+        boolean isCopied = FileHelper.copyFile(source, target);
+        if (!isCopied) {
+            log.error("Failed to copy computed.json file to /output [chainTaskId:{}]", chainTaskId);
+            return false;
+        }
+        // encrypt result if needed
+        if (taskDescription.isResultEncryption() && !encryptResult(chainTaskId)) {
```

Comment:
Probably not for today but wondering if it wouldn't be better to for E2E workflow to encrypt from /output/iexec_out/ and only copy the encrypted result to /output? I feel we might consider the /final_output (/output) as a ready-to-ship folder, avoiding to leak anything we wouldn't.

Reply:
Yes ideally we should do that. Honestly, I didn't spend time on this part I only moved the code. But yes we need to revisit it when we have time.